cmd/soroban-rpc/lib/preflight/src/fees.rs

```diff
@@ -140,23 +141,74 @@ fn calculate_soroban_resources(
     })
 }
 
-fn get_fee_configuration(_snapshot_source: &ledger_storage::LedgerStorage) -> FeeConfiguration {
-    // TODO: (at least part of) these values should be obtained from the network's ConfigSetting LedgerEntries
-    //       (instead of hardcoding them to the initial values in the network)
-    //       Specifically, we need to derive it from ConfigSettingContractComputeV0 which can
-    //       be retrieved using the ConfigSetting/CONFIG_SETTING_CONTRACT_COMPUTE_V0.
+fn get_configuration_setting(
+    ledger_storage: &ledger_storage::LedgerStorage,
+    setting_id: ConfigSettingId,
+) -> Result<ConfigSettingEntry, Box<dyn error::Error>> {
+    let key = LedgerKey::ConfigSetting(LedgerKeyConfigSetting {
+        config_setting_id: setting_id,
+    });
+    match ledger_storage.get(&key)? {
```

Comment:
We could refactor ledger_storage so `get` can fetch multiple keys at the same time. Might reduce the go<->rust round-trips. But we should probably only do that if it proves to be an issue.

Reply:
Do we have a good way to benchmark this given the rust/go boundary stuff? would a go benchmark be reliable?